EDK-II-Development-Process.md

```diff
@@ -1,297 +1,222 @@
 
-    -  To have git prompt you to selectively stage changes: `$ git add -p`
+## Contributor Process
 
-5.  Follow the commit message template given below when writing commit
-    messages
+1. Create and checkout a topic branch for your change.
+
+   `>$ git checkout -b <new-dev-branch> origin/master`
+
+2. Make changes in the working tree.
+
+3. Break up working tree changes into independent commits that do not break *git bisect*.
+   - [Commit-Partitioning](Commit-Partitioning "wikilink")
+
+   - To stage all modifications: `>$ git add -u`
+   - To add new files: `>$ git add <path-to-new-file>`
+   - To have git prompt you to selectively stage changes: `>$ git add -p`
+
+4. Follow the commit message template given below when writing commit messages.
 
     - [Commit-Message-Format](Commit-Message-Format "wikilink")
 
-    - To commit staged changes: `$ git commit`
+    - To commit staged changes: `>$ git commit`
+
+      - Tip: Add the `-s` parameter to automatically append your `Signed-off-by` tag to the commit message.
 
-      - Add the `-s` parameter to automatically append your
-        Signed-off-by tag to the commit message.
+    - Note any [TianoCore Bugzilla](https://bugzilla.tianocore.org/) items in the commit message.
 
-6.  Use the ‘PatchCheck.py’ script under ‘edk2\\BaseTools\\Scripts’
-    directory to verify the commits are correctly formatted
+5. Use the `PatchCheck.py` script under `edk2/BaseTools/Scripts` directory to verify the commits are correctly
+   formatted.
 
-    - To check the latest <N> changes: `$ python BaseTools/Scripts/PatchCheck.py -<N>`
+    - To check the latest `<N>` changes: `>$ python BaseTools/Scripts/PatchCheck.py -<N>`
 
-      - For example, 2 changes would be: `$ python BaseTools/Scripts/PatchCheck.py -2`
+      - For example, 2 changes would be: `>$ python BaseTools/Scripts/PatchCheck.py -2`
 
-    - It is strongly recommended that you run PatchCheck.py after each
-      commit. You can then easily amend the commit to correct any
-      issues.
+    - It is strongly recommended that you run `PatchCheck.py` after each commit. You can then easily amend the commit
+      to correct any issues.
 
-7.  Get the latest changes from origin
+6. Get the latest changes from the remote (`origin`).
 
-    `$ git fetch origin`
+    `>$ git fetch origin`
 
-    Note: This updates origin/master, but not your local master
-    branch. (origin/master may have newer commits than master.)
+    > Note: This updates `origin/master`, but not your local `master` branch. (`origin/master` may have newer commits
+    > than `master`).
 
-8.  Rebase the topic branch onto master branch
+7. Rebase the topic branch onto `master` branch.
 
-    `$ git rebase origin/master`
+    `>$ git rebase origin/master`
 
-9. Run the automated code formatting tool (Uncrustify) against your changes
+8. Run the automated code formatting tool (Uncrustify) against your changes.

```

Comment:
- Conditionalize the Stuart stuff on a PR status check failure.
- Mention that local build might not exactly match the CI configuration (e.g. toolchain).

Reply:
Done